src/validation.cpp

```diff
@@ -53,11 +53,60 @@
 /**
  * Global state
  */
+namespace {
+struct CBlockIndexWorkComparator
+{
+    bool operator()(CBlockIndex *pa, CBlockIndex *pb) const {
+        // First sort by most total work, ...
+        if (pa->nChainWork > pb->nChainWork) return false;
+        if (pa->nChainWork < pb->nChainWork) return true;
+
+        // ... then by earliest time received, ...
+        if (pa->nSequenceId < pb->nSequenceId) return false;
+        if (pa->nSequenceId > pb->nSequenceId) return true;
+
+        // Use pointer address as tie breaker (should only happen with blocks
+        // loaded from disk, as those all have id 0).
+        if (pa < pb) return false;
+        if (pa > pb) return true;
+
+        // Identical blocks.
+        return false;
+    }
+};
+};
+
+/**
+ * CChainState stores and provides an API to update our local copy of the chain
+ *
+ * It generally provides access to the current chain tree, as well as functions
+ * to provide it new data, which it will appropriately validate and incorporate
+ * in its state as neccessary.
+ *
+ * Eventually, the API here is targeted at being exposed externally as a
+ * consumable libconsensus library, so any functions added must only call
+ * external functions via callbacks, or to read/write data from disk.
```

Comment:
Thread https://github.com/bitcoin/bitcoin/pull/10279#discussion_r114388186

Newer comment got reverted somehow. It was:

```
/**
 * CChainState stores and provides an API to update our local knowledge of the
 * current best chain and header tree.
 *
 * It generally provides access to the current block tree, as well as functions
 * to provide new data, which it will appropriately validate and incorporate in
 * its state as necessary.
 *
 * Eventually, the API here is targeted at being exposed externally as a
 * consumable libconsensus library, so any functions added must only call
 * other class member functions, pure functions in other parts of the consensus
 * library, callbacks, or read/write-to-disk functions (eventually this will
 * also be via callbacks).
 */
```